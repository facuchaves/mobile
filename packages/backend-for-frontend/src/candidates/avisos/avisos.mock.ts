export const searchDTOResponseExample = {
  number: 0,
  size: 7,
  total: 5908,
  content: [
    {
      titulo: 'RRHH',
      tipoTrabajo: 'Full-time',
      logoURL: 'http://imgbum.prepro.jobscdn.com/portal/img/empresas/1/static/logoMainPic_13075667_bum_vd40f98b7.jpg',
      empresa: 'NavarroARG',
      localizacion: 'Ituzaingó, Buenos Aires',
      fechaPublicacion: '08-08-2019',
      salarioMinimo: 0,
      salarioMaximo: 0,
      aptoDiscapacitado: false,
      planPublicacion: {
        id: 10,
        nombre: 'Home',
      },
      confidencial: false,
      portal: 'Bumeran',
      oportunidad: false,
      fechaHoraPublicacion: '08-08-2019 11:37:49',
      idEmpresa: 13075667,
      links: [
        {
          rel: 'self',
          href: 'http://bm-applicants.prepro.api.core.jobs.navent.biz/v0/application/avisos/1113483256',
        },
      ],
      id: 1113483256,
    },
    {
      titulo: 'Asistente de Gerencia',
      tipoTrabajo: 'Full-time',
      logoURL:
        'http://imgbum.prepro.jobscdn.com/portal/img/empresas/1/static/logoMainPic_13075667_aviso_1113483255_bum_vaf337268.jpg',
      empresa: 'Iunigo',
      localizacion: 'Ituzaingó, Buenos Aires',
      fechaPublicacion: '08-08-2019',
      salarioMinimo: 0,
      salarioMaximo: 0,
      aptoDiscapacitado: false,
      planPublicacion: {
        id: 10,
        nombre: 'Home',
      },
      confidencial: false,
      portal: 'Bumeran',
      oportunidad: false,
      fechaHoraPublicacion: '08-08-2019 11:31:43',
      idEmpresa: 13075667,
      links: [
        {
          rel: 'self',
          href: 'http://bm-applicants.prepro.api.core.jobs.navent.biz/v0/application/avisos/1113483255',
        },
      ],
      id: 1113483255,
    },
    {
      titulo: 'Ejecutivo de Ventas',
      tipoTrabajo: 'Full-time',
      logoURL:
        'http://imgbum.prepro.jobscdn.com/portal/img/empresas/1/static/logoMainPic_13075667_aviso_1113483254_bum_v70f3971b.jpg',
      empresa: 'Big Blue',
      localizacion: 'Ituzaingó, Buenos Aires',
      fechaPublicacion: '08-08-2019',
      salarioMinimo: 0,
      salarioMaximo: 0,
      aptoDiscapacitado: false,
      planPublicacion: {
        id: 40,
        nombre: 'Destacados',
      },
      confidencial: false,
      portal: 'Bumeran',
      oportunidad: false,
      fechaHoraPublicacion: '08-08-2019 11:30:33',
      idEmpresa: 13075667,
      links: [
        {
          rel: 'self',
          href: 'http://bm-applicants.prepro.api.core.jobs.navent.biz/v0/application/avisos/1113483254',
        },
      ],
      id: 1113483254,
    },
    {
      titulo: 'Asistente de Gerencia',
      tipoTrabajo: 'Full-time',
      logoURL:
        'http://imgbum.prepro.jobscdn.com/portal/img/empresas/1/static/logoMainPic_13075667_aviso_1113483252_bum_v629c36d4.jpg',
      empresa: 'Médicos sin fronteras',
      localizacion: 'Ituzaingó, Buenos Aires',
      fechaPublicacion: '08-08-2019',
      salarioMinimo: 0,
      salarioMaximo: 0,
      aptoDiscapacitado: false,
      planPublicacion: {
        id: 10,
        nombre: 'Home',
      },
      confidencial: false,
      portal: 'Bumeran',
      oportunidad: false,
      fechaHoraPublicacion: '08-08-2019 11:25:01',
      idEmpresa: 13075667,
      links: [
        {
          rel: 'self',
          href: 'http://bm-applicants.prepro.api.core.jobs.navent.biz/v0/application/avisos/1113483252',
        },
      ],
      id: 1113483252,
    },
    {
      titulo: 'Call Center',
      tipoTrabajo: 'Full-time',
      logoURL:
        'http://imgbum.prepro.jobscdn.com/portal/img/empresas/1/static/logoMainPic_13075667_aviso_1113483251_bum_v914330ff.jpg',
      empresa: 'Solventa',
      localizacion: 'Ituzaingó, Buenos Aires',
      fechaPublicacion: '08-08-2019',
      salarioMinimo: 0,
      salarioMaximo: 0,
      aptoDiscapacitado: false,
      planPublicacion: {
        id: 40,
        nombre: 'Destacados',
      },
      confidencial: false,
      portal: 'Bumeran',
      oportunidad: false,
      fechaHoraPublicacion: '08-08-2019 11:23:53',
      idEmpresa: 13075667,
      links: [
        {
          rel: 'self',
          href: 'http://bm-applicants.prepro.api.core.jobs.navent.biz/v0/application/avisos/1113483251',
        },
      ],
      id: 1113483251,
    },
    {
      titulo: 'Vendedor Telefonico',
      tipoTrabajo: 'Full-time',
      logoURL: 'http://imgbum.prepro.jobscdn.com/portal/img/empresas/1/static/logoMainPic_13075667_bum_vd40f98b7.jpg',
      empresa: 'NavarroARG',
      localizacion: 'Ituzaingó, Buenos Aires',
      fechaPublicacion: '08-08-2019',
      salarioMinimo: 0,
      salarioMaximo: 0,
      aptoDiscapacitado: false,
      planPublicacion: {
        id: 10,
        nombre: 'Home',
      },
      confidencial: false,
      portal: 'Bumeran',
      oportunidad: false,
      fechaHoraPublicacion: '08-08-2019 11:21:07',
      idEmpresa: 13075667,
      links: [
        {
          rel: 'self',
          href: 'http://bm-applicants.prepro.api.core.jobs.navent.biz/v0/application/avisos/1113483249',
        },
      ],
      id: 1113483249,
    },
    {
      titulo: 'Administracion',
      tipoTrabajo: 'Full-time',
      logoURL: 'http://imgbum.prepro.jobscdn.com/portal/img/empresas/1/static/logoMainPic_13075667_bum_vd40f98b7.jpg',
      empresa: 'NavarroARG',
      localizacion: 'Ituzaingó, Buenos Aires',
      fechaPublicacion: '08-08-2019',
      salarioMinimo: 0,
      salarioMaximo: 0,
      aptoDiscapacitado: false,
      planPublicacion: {
        id: 40,
        nombre: 'Destacados',
      },
      confidencial: false,
      portal: 'Bumeran',
      oportunidad: false,
      fechaHoraPublicacion: '08-08-2019 11:20:01',
      idEmpresa: 13075667,
      links: [
        {
          rel: 'self',
          href: 'http://bm-applicants.prepro.api.core.jobs.navent.biz/v0/application/avisos/1113483248',
        },
      ],
      id: 1113483248,
    },
  ],
}
