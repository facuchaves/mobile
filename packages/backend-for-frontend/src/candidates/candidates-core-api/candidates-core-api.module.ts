import { Module, HttpModule } from '@nestjs/common'
import { CandidatesCoreApiService } from './candidates-core-api.service'
import { CoreApiModule } from '../../core-api/core-api.module'
import { ConfigModule } from '../../config/config.module'

@Module({
  imports: [ConfigModule, HttpModule, CoreApiModule],
  providers: [CandidatesCoreApiService],
  exports: [CandidatesCoreApiService],
})
export class CandidatesCoreApiModule {}
