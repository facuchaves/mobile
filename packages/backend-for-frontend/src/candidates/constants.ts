export const PREFIX_CURRICULUM = 'datosCurriculum'