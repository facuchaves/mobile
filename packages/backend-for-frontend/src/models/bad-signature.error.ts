export default class BadSignatureError extends Error {
  constructor(message) {
    super(message)
  }
}
