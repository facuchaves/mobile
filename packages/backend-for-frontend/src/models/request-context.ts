import { Site } from '../interface/site.interface'
import Oauth from './oauth'
import { Logger } from '../logger/logger.service'
import { isEmpty } from 'lodash'

export class RequestContext {
  site: Site
  oauthApplication: Oauth
  currentOauth: Oauth
  oauthPostulante: Oauth
  logger: Logger
  sessionId: any

  isCandidateLogged(): boolean {
    return !isEmpty(this.oauthPostulante)
  }
}
