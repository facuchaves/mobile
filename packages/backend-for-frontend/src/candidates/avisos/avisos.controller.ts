import {
  Controller,
  Get,
  Post,
  Param,
  Body,
  UseInterceptors,
  CacheInterceptor,
  Query,
  Req,
  Header,
} from '@nestjs/common'
import { AvisoSearchRes } from './dto/avisos-search-res.dto'
import { AvisosService } from './avisos.service'
import { ApiUseTags } from '@nestjs/swagger'
import { Context } from '../../decorators/request-context.decorator'
import { RequestContext } from '../../models/request-context'
import { AvisosSearchBodyReqCore } from './dto/avisos-search-body-req-core.dto'
import { avisosSeachParamsReq } from './dto/avisos-search-params-req.dto'
import { RecomendarAvisosDto } from './dto/recomendar-avisos.dto'
import { ApplicationAvisosService } from '../../application-services/application-avisos.service'
import { ConfigService } from '../../config/config.service'
import { PostulanteAvisoSearchResCore } from '../../candidates/avisos/dto/avisos-postulante-res-core.dto'

@ApiUseTags('avisos-avisos')
@Controller('avisos')
export class AvisosController {
  constructor(
    private readonly service: AvisosService,
    private readonly applicationAvisosService: ApplicationAvisosService,
    private readonly configService: ConfigService
  ) {}
  @Post('/search')
  async search(
    @Context() context: RequestContext,
    @Query() query: avisosSeachParamsReq,
    @Body() filterData?: AvisosSearchBodyReqCore
  ): Promise<AvisoSearchRes> {
    return this.service.search(context, { ...query, filterData })
  }

  @Get('/recommender')
  async recommender(
    @Context() context: RequestContext,
    @Query() query: RecomendarAvisosDto
  ): Promise<PostulanteAvisoSearchResCore[]> {
    const [avisosIds, estrategia] = await this.service.recomendar(context, query)
    const CANTIDAD_CHAR_DESCRIPCION_AVISO = this.configService.get('CANTIDAD_CHAR_DESCRIPCION_AVISO')
    let avisosRelacionados = await this.applicationAvisosService.list(
      context,
      avisosIds,
      CANTIDAD_CHAR_DESCRIPCION_AVISO
    )

    if (query.avisoId) {
      avisosRelacionados = avisosRelacionados.filter(it => it.id != query.avisoId).slice(0, 6)
    }

    return avisosRelacionados
  }
}
