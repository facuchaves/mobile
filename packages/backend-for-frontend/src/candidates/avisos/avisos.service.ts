import { Injectable, HttpService } from '@nestjs/common'
import { AvisoSearchRes } from './dto/avisos-search-res.dto'
import { AvisoSearchResCore } from './dto/avisos-search-res-core.dto'
import { RequestContext } from '../../models/request-context'
import { AvisosSearchBodyReqCore } from './dto/avisos-search-body-req-core.dto'
import { ApplicationCoreApiService } from '../../application-core-api/application-core-api.service'
import { ConfigService } from '../../config/config.service'
import axios from 'axios'
import { RecomendarAvisosDto } from './dto/recomendar-avisos.dto'
import { JwtService } from '@nestjs/jwt'

@Injectable()
export class AvisosService {
  constructor(
    private readonly jwtService: JwtService,
    private readonly applicationCoreApiService: ApplicationCoreApiService,
    private http: HttpService,
    private readonly configService: ConfigService
  ) {}

  async search(
    context: RequestContext,
    {
      pageSize = 20,
      page = 0,
      filterData = {},
      sort = 'DESC',
    }: { pageSize?: number; page?: number; filterData?: AvisosSearchBodyReqCore; sort?: string } = {}
  ): Promise<AvisoSearchRes> {
    const { idPais } = context.site
    const data = await this.applicationCoreApiService.post<AvisoSearchResCore>(
      context,
      `/v0/application/avisos/search?paisId=${idPais}&pageSize=${pageSize}&page=${page}&sort=${sort}`,
      filterData
    )
    const { number, size, total, content, filters } = data
    return { number, size, total, content, filters } as AvisoSearchRes
  }

  /**
   * Devuelve listado de ids de avisos recomendados (llama a API de Labs).
   * Si hay un usuario postulante logueado, entonces se utiliza idPostulante y idAviso para el
   * llamado a la API. 
   * Si no hay postulante logueado, se utiliza solamente idAviso para el llamado
   *
   * @param context
   */
  async recomendar(context: RequestContext, { avisoId, limit, modulo }: RecomendarAvisosDto) {
    let respuesta
    let estrategia

    let usuarioId = null

    if (context.oauthPostulante) {
      const accessToken = this.jwtService.decode(context.oauthPostulante.accessToken) as {
        [key: string]: any
      }
      usuarioId = accessToken ? accessToken.user_id : null
    }

    let headers = {} as any
    headers.ApplicationName = 'BUM-frontend'
    if (modulo) {
      headers.Module = modulo
    }
    let pais = context.site.sufijoPais

    const BASE_URL = this.configService.get('REST_API_RECOMMENDER_URL')
    const NOMBRE_PORTAL = this.configService.get('REST_API_RECOMMENDER_NOMBRE_PORTAL')
    const method = 
      usuarioId ? `${BASE_URL}/recommender/${NOMBRE_PORTAL}/${pais}?user=${usuarioId}&item=${avisoId}&limit=${limit}` :
      `${BASE_URL}/recommender/${NOMBRE_PORTAL}/${pais}?item=${avisoId}&limit=${limit}`

    respuesta = await axios.get(method).then(e => {
      estrategia = e.headers.strategy
      console.log(e.headers)
      return e.data
    })
  
    return [respuesta, estrategia]
  }

}
