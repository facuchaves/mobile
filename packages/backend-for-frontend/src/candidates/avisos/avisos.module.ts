import { Module, HttpModule, CacheModule } from '@nestjs/common'
import { AvisosController } from './avisos.controller'
import { AvisosService } from './avisos.service'
import { RedisModule } from '../../cache/cache-redis/redis.module'
import { CandidatesCoreApiModule } from '../candidates-core-api/candidates-core-api.module'
import { ApplicationCoreApiModule } from '../../application-core-api/application-core-api.module'
import { ApplicationAvisosModule } from '../../application-services/application-avisos.module'
import { AuthModule } from '../../auth/auth.module'

@Module({
  imports: [
    RedisModule,
    CandidatesCoreApiModule,
    ApplicationAvisosModule,
    AuthModule,
    ApplicationCoreApiModule,
    HttpModule,
  ],
  controllers: [AvisosController],
  providers: [AvisosService],
  exports: [AvisosService],
})
export class AvisosModule {}
