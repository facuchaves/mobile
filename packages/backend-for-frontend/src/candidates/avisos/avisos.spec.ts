import { AvisosService } from './avisos.service'
import { RequestContext } from '../../models/request-context'
import { CandidatesCoreApiService } from '../candidates-core-api/candidates-core-api.service'
import { searchDTOResponseExample } from './avisos.mock'
import { CandidatesService } from '../../candidates/services/candidates.service'
import { JwtService } from '@nestjs/jwt'
import { RedisService } from '../../cache/cache-redis/redis.service'

describe('Aviso de Candidates', () => {
  let service
  let candidatesCoreApiService
  let jwtService
  let httpService
  let configService

  beforeEach(async () => {
    candidatesCoreApiService = ({ post: () => {} } as unknown) as CandidatesCoreApiService
    jwtService = ({ post: () => {} } as unknown) as JwtService
    service = new AvisosService(jwtService, candidatesCoreApiService, httpService, configService)
  })

  it('should get all avisos data', async () => {
    const contextMocked = ({
      isCandidateLogged: () => true,
      site: {
        idPais: 1,
      },
    } as unknown) as RequestContext

    jest.spyOn(candidatesCoreApiService, 'post').mockImplementation(() => Promise.resolve(searchDTOResponseExample))

    const result = await service.search(contextMocked)

    expect(result).toMatchSnapshot()
  })
})
