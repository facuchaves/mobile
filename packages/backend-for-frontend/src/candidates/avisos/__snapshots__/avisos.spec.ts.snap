// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aviso de Candidates should get all avisos data 1`] = `
Object {
  "content": Array [
    Object {
      "aptoDiscapacitado": false,
      "confidencial": false,
      "empresa": "NavarroARG",
      "fechaHoraPublicacion": "08-08-2019 11:37:49",
      "fechaPublicacion": "08-08-2019",
      "id": 1113483256,
      "idEmpresa": 13075667,
      "links": Array [
        Object {
          "href": "http://bm-applicants.prepro.api.core.jobs.navent.biz/v0/application/avisos/1113483256",
          "rel": "self",
        },
      ],
      "localizacion": "Ituzaingó, Buenos Aires",
      "logoURL": "http://imgbum.prepro.jobscdn.com/portal/img/empresas/1/static/logoMainPic_13075667_bum_vd40f98b7.jpg",
      "oportunidad": false,
      "planPublicacion": Object {
        "id": 10,
        "nombre": "Home",
      },
      "portal": "Bumeran",
      "salarioMaximo": 0,
      "salarioMinimo": 0,
      "tipoTrabajo": "Full-time",
      "titulo": "RRHH",
    },
    Object {
      "aptoDiscapacitado": false,
      "confidencial": false,
      "empresa": "Iunigo",
      "fechaHoraPublicacion": "08-08-2019 11:31:43",
      "fechaPublicacion": "08-08-2019",
      "id": 1113483255,
      "idEmpresa": 13075667,
      "links": Array [
        Object {
          "href": "http://bm-applicants.prepro.api.core.jobs.navent.biz/v0/application/avisos/1113483255",
          "rel": "self",
        },
      ],
      "localizacion": "Ituzaingó, Buenos Aires",
      "logoURL": "http://imgbum.prepro.jobscdn.com/portal/img/empresas/1/static/logoMainPic_13075667_aviso_1113483255_bum_vaf337268.jpg",
      "oportunidad": false,
      "planPublicacion": Object {
        "id": 10,
        "nombre": "Home",
      },
      "portal": "Bumeran",
      "salarioMaximo": 0,
      "salarioMinimo": 0,
      "tipoTrabajo": "Full-time",
      "titulo": "Asistente de Gerencia",
    },
    Object {
      "aptoDiscapacitado": false,
      "confidencial": false,
      "empresa": "Big Blue",
      "fechaHoraPublicacion": "08-08-2019 11:30:33",
      "fechaPublicacion": "08-08-2019",
      "id": 1113483254,
      "idEmpresa": 13075667,
      "links": Array [
        Object {
          "href": "http://bm-applicants.prepro.api.core.jobs.navent.biz/v0/application/avisos/1113483254",
          "rel": "self",
        },
      ],
      "localizacion": "Ituzaingó, Buenos Aires",
      "logoURL": "http://imgbum.prepro.jobscdn.com/portal/img/empresas/1/static/logoMainPic_13075667_aviso_1113483254_bum_v70f3971b.jpg",
      "oportunidad": false,
      "planPublicacion": Object {
        "id": 40,
        "nombre": "Destacados",
      },
      "portal": "Bumeran",
      "salarioMaximo": 0,
      "salarioMinimo": 0,
      "tipoTrabajo": "Full-time",
      "titulo": "Ejecutivo de Ventas",
    },
    Object {
      "aptoDiscapacitado": false,
      "confidencial": false,
      "empresa": "Médicos sin fronteras",
      "fechaHoraPublicacion": "08-08-2019 11:25:01",
      "fechaPublicacion": "08-08-2019",
      "id": 1113483252,
      "idEmpresa": 13075667,
      "links": Array [
        Object {
          "href": "http://bm-applicants.prepro.api.core.jobs.navent.biz/v0/application/avisos/1113483252",
          "rel": "self",
        },
      ],
      "localizacion": "Ituzaingó, Buenos Aires",
      "logoURL": "http://imgbum.prepro.jobscdn.com/portal/img/empresas/1/static/logoMainPic_13075667_aviso_1113483252_bum_v629c36d4.jpg",
      "oportunidad": false,
      "planPublicacion": Object {
        "id": 10,
        "nombre": "Home",
      },
      "portal": "Bumeran",
      "salarioMaximo": 0,
      "salarioMinimo": 0,
      "tipoTrabajo": "Full-time",
      "titulo": "Asistente de Gerencia",
    },
    Object {
      "aptoDiscapacitado": false,
      "confidencial": false,
      "empresa": "Solventa",
      "fechaHoraPublicacion": "08-08-2019 11:23:53",
      "fechaPublicacion": "08-08-2019",
      "id": 1113483251,
      "idEmpresa": 13075667,
      "links": Array [
        Object {
          "href": "http://bm-applicants.prepro.api.core.jobs.navent.biz/v0/application/avisos/1113483251",
          "rel": "self",
        },
      ],
      "localizacion": "Ituzaingó, Buenos Aires",
      "logoURL": "http://imgbum.prepro.jobscdn.com/portal/img/empresas/1/static/logoMainPic_13075667_aviso_1113483251_bum_v914330ff.jpg",
      "oportunidad": false,
      "planPublicacion": Object {
        "id": 40,
        "nombre": "Destacados",
      },
      "portal": "Bumeran",
      "salarioMaximo": 0,
      "salarioMinimo": 0,
      "tipoTrabajo": "Full-time",
      "titulo": "Call Center",
    },
    Object {
      "aptoDiscapacitado": false,
      "confidencial": false,
      "empresa": "NavarroARG",
      "fechaHoraPublicacion": "08-08-2019 11:21:07",
      "fechaPublicacion": "08-08-2019",
      "id": 1113483249,
      "idEmpresa": 13075667,
      "links": Array [
        Object {
          "href": "http://bm-applicants.prepro.api.core.jobs.navent.biz/v0/application/avisos/1113483249",
          "rel": "self",
        },
      ],
      "localizacion": "Ituzaingó, Buenos Aires",
      "logoURL": "http://imgbum.prepro.jobscdn.com/portal/img/empresas/1/static/logoMainPic_13075667_bum_vd40f98b7.jpg",
      "oportunidad": false,
      "planPublicacion": Object {
        "id": 10,
        "nombre": "Home",
      },
      "portal": "Bumeran",
      "salarioMaximo": 0,
      "salarioMinimo": 0,
      "tipoTrabajo": "Full-time",
      "titulo": "Vendedor Telefonico",
    },
    Object {
      "aptoDiscapacitado": false,
      "confidencial": false,
      "empresa": "NavarroARG",
      "fechaHoraPublicacion": "08-08-2019 11:20:01",
      "fechaPublicacion": "08-08-2019",
      "id": 1113483248,
      "idEmpresa": 13075667,
      "links": Array [
        Object {
          "href": "http://bm-applicants.prepro.api.core.jobs.navent.biz/v0/application/avisos/1113483248",
          "rel": "self",
        },
      ],
      "localizacion": "Ituzaingó, Buenos Aires",
      "logoURL": "http://imgbum.prepro.jobscdn.com/portal/img/empresas/1/static/logoMainPic_13075667_bum_vd40f98b7.jpg",
      "oportunidad": false,
      "planPublicacion": Object {
        "id": 40,
        "nombre": "Destacados",
      },
      "portal": "Bumeran",
      "salarioMaximo": 0,
      "salarioMinimo": 0,
      "tipoTrabajo": "Full-time",
      "titulo": "Administracion",
    },
  ],
  "filters": undefined,
  "number": 0,
  "size": 7,
  "total": 5908,
}
`;
